test/test_images/readiness/readiness.go

```diff
@@ -87,7 +76,22 @@ func main() {
 		time.Sleep(delay)
 	}
 
-	http.ListenAndServe(":8080", nil)
+	mainServer := http.NewServeMux()
+	mainServer.HandleFunc("/", handleMain)
+	mainServer.HandleFunc("/start-failing", handleStartFailing)
+
+	probeServer := http.NewServeMux()
+	probeServer.HandleFunc("/", handleHealthz)
+
+	go func() {
+		http.ListenAndServe(":"+strconv.Itoa(healthcheckPort), probeServer)
```

Comment:
if the health check port isn't set we want it on the mainServer

That's probably the cause of the failures

Reply:
Or you could add the `handleHealthz` handler to _both_ servers to keep things simple